chore(deps): Adding exporter for rutorrent

ac1dburnz opened the pull request:
**Description**
<!--
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
-->
⚒️ Fixes  # <!--(issue)-->

**⚙️ Type of change**

- [ ] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**
<!--
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
-->

**📃 Notes:**
<!-- Please enter any other relevant information here -->

**✔️ Checklist:**

- [ ] ⚖️ My code follows the style guidelines of this project
- [ ] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

ac1dburnz committed:
chore(deps): Adding exporter for torrent

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz committed:
Update questions.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz committed:
Update values.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz committed:
Update Chart.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz committed:
Merge branch 'master' into patch-5

ac1dburnz committed:
Update values.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz committed:
Update Chart.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz committed:
Update common.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz committed:
Update values.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz committed:
Update values.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz committed:
Update values.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz committed:
Update values.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz committed:
Update values.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz commented:
sidecart looks to be running on my test instance correctly 

<img width="1386" alt="Screenshot 2024-02-01 at 12 57 31 AM" src="https://github.com/truecharts/charts/assets/10972668/03be38f3-8e3a-40e8-a683-f26abfad6743">

ac1dburnz committed:
Update values.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz commented:
might have spoken too soon

024-02-01 01:05:49.282245-05:00level=info ts=2024-02-01T06:05:49.282Z caller=main.go:38 build_context="(go=go1.21.5, platform=linux/amd64, user=, date=2023-12-18T15:03:50Z, tags=unknown)"
2024-02-01 01:05:49.282386-05:00level=info ts=2024-02-01T06:05:49.282Z caller=tls_config.go:313 msg="Listening on" address=[::]:9135
2024-02-01 01:05:49.282406-05:00level=info ts=2024-02-01T06:05:49.282Z caller=tls_config.go:316 msg="TLS is disabled." http2=false address=[::]:9135
2024-02-01 01:06:17.623326-05:00level=error ts=2024-02-01T06:06:17.623Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:06:32.614683-05:00level=error ts=2024-02-01T06:06:32.614Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:06:47.614490-05:00level=error ts=2024-02-01T06:06:47.614Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:07:02.613855-05:00level=error ts=2024-02-01T06:07:02.613Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:07:17.614125-05:00level=error ts=2024-02-01T06:07:17.614Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:07:32.614350-05:00level=error ts=2024-02-01T06:07:32.614Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:07:47.613899-05:00level=error ts=2024-02-01T06:07:47.613Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:08:02.614166-05:00level=error ts=2024-02-01T06:08:02.614Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:08:17.614354-05:00level=error ts=2024-02-01T06:08:17.614Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:08:32.614877-05:00level=error ts=2024-02-01T06:08:32.614Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:08:47.614789-05:00level=error ts=2024-02-01T06:08:47.614Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:09:02.614140-05:00level=error ts=2024-02-01T06:09:02.614Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:09:17.614770-05:00level=error ts=2024-02-01T06:09:17.614Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:09:32.614624-05:00level=error ts=2024-02-01T06:09:32.614Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:09:47.614746-05:00level=error ts=2024-02-01T06:09:47.614Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:10:02.614758-05:00level=error ts=2024-02-01T06:10:02.614Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:10:17.613731-05:00level=error ts=2024-02-01T06:10:17.613Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:10:32.614563-05:00level=error ts=2024-02-01T06:10:32.614Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:10:47.615029-05:00level=error ts=2024-02-01T06:10:47.614Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:11:02.614381-05:00level=error ts=2024-02-01T06:11:02.614Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
2024-02-01 01:11:17.614502-05:00level=error ts=2024-02-01T06:11:17.614Z caller=exporter.go:47 msg="Can't scrape rTorrent" err="system.hostname XMLRPC call failed: POST failed: Post \"http://0.0.0.0:8000/RPC2\": dial tcp 0.0.0.0:8000: connect: connection refused"
failed to create fsnotify watcher: too many open files

ac1dburnz committed:
Update values.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz commented:
Interesting findings,
I tried 127.0.0.1 , 0.0.0.0 and the dns adress and none of them produced any data for Prometheus. When I directly used the ip address of the app that i manually added, it worked and i am receiving data. 

<img width="571" alt="Screenshot 2024-02-01 at 2 09 58 AM" src="https://github.com/truecharts/charts/assets/10972668/7808c336-e8e1-44c9-9e29-c07813498f14">

Further examples:
<img width="772" alt="Screenshot 2024-02-01 at 2 13 07 AM" src="https://github.com/truecharts/charts/assets/10972668/a8f8d134-bf85-4a0f-beb8-c00ee89c675b">

ac1dburnz committed:
Update values.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

xstar97 commented on charts/stable/rtorrent-rutorrent/values.yaml:
```diff
@@ -78,16 +92,61 @@ workload:
+          enabled: true
+          imageSelector: exportarrImage
+          args:
+            - "--rtorrent.scrape-uri=http://127.0.0.1:8000/RPC2/"
```
```suggestion
            - --rtorrent.scrape-uri='{{ printf "http://%v:%v/RPC2/" (include "tc.v1.common.lib.chart.names.fullname" $) .Values.service.main.ports.main.port }}'
```

ac1dburnz committed:
Update charts/stable/rtorrent-rutorrent/values.yaml

Co-authored-by: Xstar97TheNoob <9399967+xstar97@users.noreply.github.com>
Signed-off-by: Ac1dburn <colinborins@hotmail.com>

ac1dburnz commented:
@xstar97  it also seems that the persistence are not mounting. Any ideas?

xstar97 commented on charts/stable/rtorrent-rutorrent/values.yaml:
```diff
@@ -78,16 +92,61 @@ workload:
+      exportarr:
+        exportarr:
+          mountPath: /config
+          readOnly: true
```
redo this section plz

```yaml
persistence:
  data:
    enabled: true
    targetSelector:
      main:
        main:
          mountPath: "/data"
      exportarr:
        exportarr:
          mountPath: "/config"
          readOnly: true
  downloads:
    enabled: true
    mountPath: "/downloads"
  passwd:
    enabled: true
    mountPath: "/passwd"
```

ac1dburnz committed:
Update values.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz committed:
Update values.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

xstar97 commented on charts/stable/rtorrent-rutorrent/values.yaml:
```diff
@@ -88,6 +129,21 @@ persistence:
+  exportarr:
+    enabled: true
+    mountPath: "/config"
+    readOnly: true
```
```suggestion
```

xstar97 replied:
this is not a thing, actually copy what i have given you or return the code back for the persistence, i see no options or code to suggest _your_ exportarr sidecar needs access to the config dir.

your confusing two separate images that have nothing to do with another.

stop.
read the upstreams docs
and fix the chart.

ac1dburnz committed:
Update values.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz committed:
Update charts/stable/rtorrent-rutorrent/values.yaml

Co-authored-by: Xstar97TheNoob <9399967+xstar97@users.noreply.github.com>
Signed-off-by: Ac1dburn <colinborins@hotmail.com>

ac1dburnz committed:
removing ''

'' was throwing an error 

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz committed:
Update values.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz commented:
Tested and good to go! 
<img width="1648" alt="Screenshot 2024-02-01 at 8 21 04 PM" src="https://github.com/truecharts/charts/
<img width="1010" alt="Screenshot 2024-02-01 at 8 20 39 PM" src="https://github.com/truecharts/charts/assets/10972668/6917be4b-6282-45c9-ba2b-0d5bd28438d9">
assets/10972668/325dd0a7-6ec9-4275-9f0e-740400fb0fbc">

I was also able to update my app and all the data stayed.

ac1dburnz committed:
Update values.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz committed:
Merge branch 'master' into patch-5

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz reviewed (commented):
Adding exporters new update

ac1dburnz committed:
Merge branch 'master' into patch-5

ac1dburnz committed:
Merge branch 'master' into patch-5

ac1dburnz committed:
Merge branch 'master' into patch-5

ac1dburnz commented:
Been using this for about a week and no issues

ac1dburnz committed:
Bug in 1.3.0 

There’s a bug in 1.3.0 where the labels stop the ability to see stopped and all torrents. 

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

Crow-Control commented on charts/stable/rtorrent-rutorrent/values.yaml:
```diff
@@ -78,6 +92,33 @@ workload:
+          enabled: true
+          imageSelector: exportarrImage
+          args:
+            - --rtorrent.scrape-uri={{ printf "http://%s-xmlrpc.%s.svc.cluster.local:%v/RPC2/" .Release.Name .Release.Namespace .Values.service.xmlrpc.ports.xmlrpc.port }}
```
```suggestion
            - --rtorrent.scrape-uri={{ printf "http://%s-xmlrpc.%s:%v/RPC2/" .Release.Name .Release.Namespace .Values.service.xmlrpc.ports.xmlrpc.port }}
```

svc.cluster.local is technically not guaranteed

Crow-Control commented on charts/stable/rtorrent-rutorrent/values.yaml:
```diff
@@ -78,6 +92,33 @@ workload:
+          env:
+            INTERFACE: 0.0.0.0
+            PORT: "{{ .Values.service.metrics.ports.metrics.port }}"
+            URL: '{{ printf "http://%s-xmlrpc.%s.svc.cluster.local:%v/RPC2/" .Release.Name .Release.Namespace .Values.service.xmlrpc.ports.xmlrpc.port }}'
```
```suggestion
            URL: '{{ printf "http://%s-xmlrpc.%s.svc.cluster.local:%v/RPC2/" .Release.Name .Release.Namespace .Values.service.xmlrpc.ports.xmlrpc.port }}' same here
```

ac1dburnz committed:
Update charts/stable/rtorrent-rutorrent/values.yaml

Co-authored-by: Kjeld Schouten <info@kjeldschouten.nl>
Signed-off-by: Ac1dburn <colinborins@hotmail.com>

ac1dburnz committed:
Update charts/stable/rtorrent-rutorrent/values.yaml

Co-authored-by: Kjeld Schouten <info@kjeldschouten.nl>
Signed-off-by: Ac1dburn <colinborins@hotmail.com>

ac1dburnz committed:
Update values.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz committed:
Update values.yaml

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz commented:
Fixed based on your comments

ac1dburnz committed:
Update to exporter

Signed-off-by: Ac1dburn <ac1dburnez@gmail.com>

ac1dburnz committed:
Merge branch 'master' into patch-5

ac1dburnz commented:
@Ornias1993  was there something missing here too?

ac1dburnz committed:
Merge branch 'master' into patch-5
